### DISCUSSION TITLE
Check friendship before retweeting.

### DISCUSSION THREAD

**Discussion by Chokitu** [tweepy/tweepy#1436] [Questions] [answered]
So, I've been trying to fix a little script that I have here.

I am able to retweet the message that I want using Cursor Search, but I wanted to add a condition that the bot profile needs to be following the person before it retweets

Let’s say person A is followed by the bot and posts something like: #RTMe
But person B is not followed by the bot
and replies the same thing, but the bot only retweets persons A tweet
Note that the bot must be following the person, not the person following the bot

I've been trying to do it with

`for page in tweepy.Cursor(api.friends, count=100).pages():
	user_ids = [user.id for user in page]`

and then checking a relation with
`relation = api.show_friendship(source_id=user.id, target_id=user_ids)`

But as I am getting the array for the user_ids I am not able to check one by one, If I am to make a for statement will probably consume too much memory, is there another way?

**Comment by Harmon758** [comment, marked as answer]
I'm not sure I understand what you're trying to do here. [`API.friends`](https://tweepy.readthedocs.io/en/latest/api.html#API.friends) already returns users you're following. Why pass it back to `API.show_friendship`? If you already have the Tweet/`Status` object, you can simply check if the ID of the user who Tweeted is in your list (using the `user` attribute of the `Status` object and the corresponding `id` attribute of that User object, e.g. `tweet.user.id`), or if you wanted to check it in realtime, use `API.show_friendship` with that user ID.

Also, if you just wanted the IDs of the users you're following, you can use [`API.friends_ids`](https://tweepy.readthedocs.io/en/latest/api.html#API.friends_ids) instead.

In any case, to answer your question, you can use [`API.lookup_friendships`](https://tweepy.readthedocs.io/en/latest/api.html#API.lookup_friendships) to bulk lookup relationships to 100 `user_ids` at a time.

**Comment by Chokitu** [comment]
It's says that 'API' object has no attribute 'lookup_friendship'

**Comment by Harmon758** [comment]
The method in question is [`API.lookup_friendships`](https://tweepy.readthedocs.io/en/latest/api.html#API.lookup_friendships) with an `s`.

**Comment by Chokitu** [comment]
Oops, sorry. I will try that right now!

**Comment by Chokitu** [comment]
```
for tweet in tweepy.Cursor(api.search, q=('#RTME')).items(500):
			try:
				
				relation = api.lookup_friendships(user_ids = tweet.user.id)
				for friendship in relation:
					if(friendship.is_following):
					    tweet.favorite()
```

Am I doing something wrong?

**Comment by Harmon758** [comment]
Like I said,  [`API.lookup_friendships`](https://tweepy.readthedocs.io/en/latest/api.html#API.lookup_friendships) can be used to bulk lookup relationships to 100 `user_ids` at a time. From the documentation, you can see that the `user_ids` parameter accepts a "list of user IDs". If you want to lookup a single relationship, you can use `API.show_friendship`, as you were attempting to do before.

For code block usage, see https://docs.github.com/en/github/writing-on-github/creating-and-highlighting-code-blocks.

**Comment by Chokitu** [comment]
I want to check for multiple relationships, tho.

**Comment by Harmon758** [comment]
You can use [`API.lookup_friendships`](https://tweepy.readthedocs.io/en/latest/api.html#API.lookup_friendships) to do so.
`tweet.user.id` is a single integer user ID here, corresponding to the ID of the user who posted each Tweet.

**Comment by Chokitu** [comment]
Oh, that makes sense. I was thinking about that.

So I can just do like:

`following = api.friends_ids()`

and then
`friendships = api.lookup_friendships(user_ids = following) `

**Comment by Harmon758** [comment]
[`API.friends_ids`](https://tweepy.readthedocs.io/en/latest/api.html#API.friends_ids) is a method, so it would be `following = api.friends_ids()`, but again, I'm not sure why you're doing this since `API.friends_ids` will already be returning IDs of users being followed.

Also note, `API.friends_ids` uses the [GET friends/ids endpoint](https://developer.twitter.com/en/docs/twitter-api/v1/accounts-and-users/follow-search-get-users/api-reference/get-friends-ids), which returns results in groups of up to 5,000 user IDs, so if you have more or are missing results, you'll need to paginate it with a `Cursor` object as well.

**Comment by Chokitu** [comment]
Yep, I was seeing that, I just created a new profile account not following anyone and tried it out. But for some reason its not working. I mean it is liking and retweeting the post. But its not checking properly if the person is being followed by me. I am trying to debug it, but not successfully so far. This is what I got

```python
for tweet in tweepy.Cursor(api.search, q=('#EuQueroMuitoTestarIsso')).items(numeroTweets):
			try:
				seguindo = api.friends_ids()
				friendships = api.lookup_friendships(user_ids = seguindo) 
				for friendship in friendships:
					print(friendship.is_following)
					if(friendship.is_following):
						tweet.favorite()
```

**Comment by Harmon758** [comment]
For each search result, this is looking up all the users you're following; looking up the relationships for those users; for each relationship, checking if you're following them in that relationship, which of course you are, since you looked up all the users you're following to begin with; and then favoriting the same Tweet for each user you're following. This is redundant in almost every way.

Again, I'm not sure what you're trying to accomplish here. Like I said initially, if you want to check that the user who Tweeted is someone you're following, you can simply use `API.show_friendship` with that user ID. Otherwise, if you want to store and use a list of IDs of users you follow and compare against that list (perhaps to avoid hitting the rate limit of querying the [GET friendships/show endpoint](https://developer.twitter.com/en/docs/twitter-api/v1/accounts-and-users/follow-search-get-users/api-reference/get-friendships-show) that `API.show_friendship` uses each time), you can use `API.friends_ids`.

**Comment by Chokitu** [comment]
Okay, I am sorry. I was very sleepy, I took a better look at the documentation and I feel really dumb now, thanks for the patience, everything is working great!
